File: 1002 - Find Common Characters.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var commonChars = function (words) {
    let count = new Map();
    for (let c of words[0]) {
        if (count.has(c)) {
            count.set(c, count.get(c) + 1);
        } else {
            count.set(c, 1);
        }
    }
    for (let i = 1; i < words.length; i++) {
        for (let c of count.keys()) {
            let patt = new RegExp(c, 'g');
            let arr = words[i].match(patt);
            count.set(c, Math.min(arr == null ? 0 : arr.length, count.get(c)));
        }
    }
    let res = [];
    for (let c of count.keys()) {
        for (let i = 0; i < count.get(c); i++) {
            res.push(c);
        }
    }
    return res;
};

File: 1002 - Find Common Characters.cpp
#include <unordered_map>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        unordered_map<char, int> count;
        for (char c : words[0]) {
            count[c]++;
        }
        for (int i = 1; i < words.size(); i++) {
            for (auto p = count.begin(); p != count.end(); p++) {
                int cnt = countc(words[i], p->first);
                count[p->first] = count[p->first] < cnt ? count[p->first] : cnt;
            }
        }
        vector<string> res;
        for (auto p = count.begin(); p != count.end(); p++) {
            for (int i = 0; i < p->second; i++) {
                res.push_back(string{p->first});
            }
        }
        return res;

    }

    int countc(string word, char c) {
        int res = 0;
        for (char a : word) {
            if (a == c) res++;
        }
        return res;
    }
};